sonic-utilities-data/templates/sonic-cli-gen/config.py.j2

```diff
@@ -0,0 +1,559 @@
+{% if source_template is defined %}
+Source template: {{ source_template }}
+{% endif %}
+{% if source_yang_module is defined %}
+Source YANG module: {{ source_yang_module }}
+{% endif %}
+"""
+
+import click
+import utilities_common.cli as clicommon
+import utilities_common.general as general
+from config import config_mgmt
+
+
+# Load sonic-cfggen from source since /usr/local/bin/sonic-cfggen does not have .py extension.
+sonic_cfggen = general.load_module_from_source('sonic_cfggen', '/usr/local/bin/sonic-cfggen')
+
+
+def exit_with_error(*args, **kwargs):
+    """ Print a message with click.secho and abort CLI.
+
+    Args:
+        args: Positional arguments to pass to click.secho
+        kwargs: Keyword arguments to pass to click.secho
+    """
+
+    click.secho(*args, **kwargs)
+    raise click.Abort()
+
+
+def validate_config_or_raise(cfg):
+    """ Validate config db data using ConfigMgmt.
+
+    Args:
+        cfg (Dict): Config DB data to validate.
+    Raises:
+        Exception: when cfg does not satisfy YANG schema.
+    """
+
+    try:
+        cfg = sonic_cfggen.FormatConverter.to_serialized(cfg)
+        config_mgmt.ConfigMgmt().loadData(cfg)
+    except Exception as err:
+        raise Exception('Failed to validate configuration: {}'.format(err))
+
+
+def add_entry_validated(db, table, key, data):
+    """ Add new entry in table and validate configuration.
+
+    Args:
+        db (swsscommon.ConfigDBConnector): Config DB connector obect.
+        table (str): Table name to add new entry to.
+        key (Union[str, Tuple]): Key name in the table.
+        data (Dict): Entry data.
+    Raises:
+        Exception: when cfg does not satisfy YANG schema.
+    """
+
+    cfg = db.get_config()
+    cfg.setdefault(table, {})
+    if key in cfg[table]:
+        raise Exception(f"{key} already exists")
+
+    cfg[table][key] = data
+
+    validate_config_or_raise(cfg)
+    db.set_entry(table, key, data)
+
+
+def update_entry_validated(db, table, key, data, create_if_not_exists=False):
+    """ Update entry in table and validate configuration.
+    If attribute value in data is None, the attribute is deleted.
+
+    Args:
+        db (swsscommon.ConfigDBConnector): Config DB connector obect.
+        table (str): Table name to add new entry to.
+        key (Union[str, Tuple]): Key name in the table.
+        data (Dict): Entry data.
+        create_if_not_exists (bool):
+            In case entry does not exists already a new entry
+            is not created if this flag is set to False and
+            creates a new entry if flag is set to True.
+    Raises:
+        Exception: when cfg does not satisfy YANG schema.
+    """
+
+    cfg = db.get_config()
+    cfg.setdefault(table, {})
+
+    if create_if_not_exists:
+        cfg[table].setdefault(key, {})
+
+    if key not in cfg[table]:
+        raise Exception(f"{key} does not exist")
+
+    for attr, value in data.items():
+        if value is None and attr in cfg[table][key]:
```

Comment:
"if value is None", as pop will throw if attr exists, else no-op, if you call as cfg[table][key].pop(attr**, None**)

Reply:
Thanks, done.